plutus-contract/src/Plutus/Contract/Request.hs

```diff
@@ -181,6 +182,23 @@ awaitTime ::
     -> Contract w s e POSIXTime
 awaitTime s = pabReq (AwaitTimeReq s) E._AwaitTimeResp
 
+-- | Await a promise, or throw an error after timeout
+withTimeout
+  :: forall w endpoints err a.
+     AsContractError err
+  => -- | Timeout duration to wait for. Note: this will round to Slot time
+     POSIXTime
```

Comment:
It would be nice if we had a different type for time spans. Because `POSIXTime` is normally an absolute point in time. Basically what the `time` library does with `NominalDiffTime`. (not for this PR though)